File: Restaurante/AppRestaurante/src/app/Pagamento.ts
export class Pagamento {

    public id: number;
    public dinheiroPagamentos: Array<Dinheiro>;
    public cartaoPagamentos: Array<Cartao>;

    public constructor(id: number) {
        this.id = id;
        this.dinheiroPagamentos = new Array<Dinheiro>();
        this.cartaoPagamentos = new Array<Cartao>();
    }
}

export class Dinheiro {

    public notas: string;

    public constructor(notas: string) {
        this.notas = notas;
    }
}

export class Cartao {

    public nomeDoDono: string;
    public numCartao: number;
    public bandeira: string;

    public constructor(nomeDoDono: string, numCartao: number, bandeira: string) {
        this.nomeDoDono = nomeDoDono;
        this.numCartao = numCartao;
        this.bandeira = bandeira;
    }
}

File: Restaurante/Models/Pagamento.cs
using System.ComponentModel.DataAnnotations;

namespace Restaurante.Models
{
    public class Pagamento
    {
        [Key]
        public int Id { get; set; }
        public List<Dinheiro> DinheiroPagamentos { get; set; }
        public List<Cartao> CartaoPagamentos { get; set; }

        public Pagamento(int id)
        {
            Id = id;
            DinheiroPagamentos = new List<Dinheiro>();
            CartaoPagamentos = new List<Cartao>();
        }

        public class Dinheiro
        {

            public string Notas { get; set; }

            public Dinheiro(string notas)
            {
                Notas = notas;
            }
        }

        public class Cartao
        {
            public string NomeDoDono { get; set; }
            public long NumCartao { get; set; }
            public string Bandeira { get; set; }

            public Cartao(string nomeDoDono, long numCartao, string bandeira)
            {
                NomeDoDono = nomeDoDono;
                NumCartao = numCartao;
                Bandeira = bandeira;
            }
        }
    }
}